docs/book/src/reference/webhook-for-core-types.md

```diff
@@ -0,0 +1,80 @@
+# Admission Webhook for Core Types
+
+It is very easy to build admission webhooks for CRDs, which has been covered in
+the CronJob tutorial. Given that kubebuilder doesn't support webhook scaffolding
+for core types, you have to use the library from controler-runtime to handle it.
+There is an [example](https://github.com/kubernetes-sigs/controller-runtime/tree/master/examples/builtins)
+in controller-runtime.
+
+It is suggested to use kubebuilder to initialize a project, and then you can
+follow the steps below to add admission webhooks for core types.
+
+## Implement Your Handler
+
+You need to have your handler implements the
+[admission.Handler](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/webhook/admission#Handler)
+interface.
+
+```go
+type podAnnotator struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+func (a *podAnnotator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	pod := &corev1.Pod{}
+	err := a.decoder.Decode(req, pod)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	// mutate the fields in pod
+
+	marshaledPod, err := json.Marshal(pod)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+}
+```
+
+If you need a client, just add the following InjectClient method for your 
```

Comment:
probably just suggest that they pass in the client by hand if they need it (at struct construction time).